app/src/main/java/com/eveningoutpost/dexdrip/webservices/WebServiceSgv.java

```diff
@@ -140,56 +191,69 @@ public WebResponse request(String query) {
+                // add these items to the cached json object
+                JSONObject origItem = reply.getJSONObject(0);
+                JSONObject item = new JSONObject(origItem.toString());
+                if (units_indicator > 0) {
+                    String unitsHint = Pref.getString("units", "mgdl").equals("mgdl") ? "mgdl" : "mmol";
+                    item.put("units_hint", unitsHint);
+                }
 
-                    // emit result code from tasker if present
-                    if (tasker_result_code > 0) {
-                        item.put("tasker_result", tasker_result_code);
-                        tasker_result_code = 0;
-                    }
+                // emit the external status line once if present
+                String external_status_line = getLastStatusLine();
+                if (external_status_line.length() > 0) {
+                    item.put("aaps", external_status_line);
+                    item.put("aaps-ts", getLastStatusLineTime());
+                }
 
-                    // emit nano-status string if present
-                    if (collector_status_string != null) {
-                        item.put("collector_status", collector_status_string);
-                        collector_status_string = null;
-                    }
+                // emit result code from steps if present
+                if (steps_result_code > 0) {
+                    item.put("steps_result", steps_result_code);
+                }
 
-                    // TODO uploader battery
+                // emit result code from heart if present
+                if (heart_result_code > 0) {
+                    item.put("heart_result", heart_result_code);
+                }
 
-                    // emit sensor status/age message if present
-                    if (sensor_status_string != null) {
-                        item.put("sensor_status", sensor_status_string);
-                        sensor_status_string = null;
-                    }
+                // emit result code from tasker if present
+                if (tasker_result_code > 0) {
+                    item.put("tasker_result", tasker_result_code);
+                }
 
-                    reply.put(item);
+                // emit nano-status string if present
+                if (collector_status_string != null) {
+                    item.put("collector_status", collector_status_string);
                 }
 
-                Log.d(TAG, "Output: " + reply.toString());
+                // TODO uploader battery
+
+                // emit sensor status/age message if present
+                if (sensor_status_string != null) {
+                    item.put("sensor_status", sensor_status_string);
+                }
+                reply.put(0, item);
             } catch (JSONException e) {
-                UserError.Log.wtf(TAG, "Got json exception: " + e);
+                e.printStackTrace();
```

Comment:
Shouldn't we inform the user if this exception occurs?

Reply:
Eh, this would be a programming bug if it ever occurred. Shouldn't actually be possible for this to happen, the JSON put method just explicitly requires that the JSONException be caught.

Reply:
But `e.printStackTrace()` is logged to adb, AFAIK. Hence, you will only see this exception during debugging. Isn't it better to use `UserError.Log.e()` instead?